src/Cms/Language.php

```diff
@@ -427,11 +435,31 @@ public function save(): static
 
 		ksort($data);
 
+		// save translations to the custom root and remove translations
+		// to prevent duplication write into the language file
+		if ($this->translationsObject()->root() !== null) {
+			$this->translationsObject()
+				->save($data['translations'] ?? []);
+			$data['translations'] = [];
+		}
+
 		Data::write($this->root(), $data);
 
 		return $this;
 	}
 
+	/**
+	 * Sets the translations data
+	 *
+	 * @return $this
+	 */
+	protected function setTranslations(array $translations = []): static
```

Comment:
Since everything else happens directly in the constructor, I wonder if we'd better off also putting this code directly in the constructor instead of adding a `setTranslations` method.